src/components/ScanQRModal/index.tsx

```diff
@@ -28,22 +38,35 @@ const ScanQRModal = ({ classes, isOpen, onClose, onScan }) => {
   useEffect(() => {
     checkWebcam(
       () => {
-        setHasWebcam(true)
+        setUseWebcam(true)
       },
       () => {
-        setHasWebcam(false)
+        setUseWebcam(false)
       },
     )
   }, [])
 
   useEffect(() => {
-    // this fires only when the hasWebcam changes to false (null > false (user doesn't have webcam)
-    // , true > false (user switched from webcam to file upload))
-    // Doesn't fire on re-render
-    if (hasWebcam === false) {
+    if (useWebcam === false && !fileUploadModalOpen && !error) {
+      setFileUploadModalOpen(true)
       openImageDialog()
     }
-  }, [hasWebcam, openImageDialog])
+  }, [useWebcam, openImageDialog, fileUploadModalOpen, setFileUploadModalOpen, error])
+
+  const onFileScannedResolve = (error: string | null, successData: string | null) => {
+    if (successData) {
+      onScan(successData)
+    }
+    if (error) {
+      console.error('Error uploading file', error)
+      setError(`The QR could not be read`)
+    }
+    if (!useWebcam) {
```

Comment:
is this validation needed? seems like if useWebcam is null or false, it will never reach this line.

Reply:
Yes because if it's false means that the user opened the modal for uploading the file and if that line was reached that means that there was not an error but the qr could not be readed (it returns null or successData) and we need to display a feedback in that case